sourced/ml/algorithms/uast_struct_to_bag.py

```diff
@@ -87,103 +100,77 @@ def __init__(self, p_explore_neighborhood, q_leave_neighborhood, n_walks, n_step
         self.q_leave_neighborhood = q_leave_neighborhood
         self.n_walks = n_walks
         self.n_steps = n_steps
-        self.type2ind = type2ind if type2ind is not None else FakeVocabulary()
+        self.node2index = node2index if node2index is not None else Node2InternalType()
+        if seed is not None:
+            random.seed(seed)
 
-    def uast2walks(self, uast):
+    def __call__(self, uast):
         starting_nodes = self.prepare_starting_nodes(uast)
-        res = []
         for i in range(self.n_walks):
             for start_node in starting_nodes:
-                res.append(self.random_walk(start_node))
-        return res
+                yield self.random_walk(start_node)
```

Comment:
An iterator is better due to next `__call__ ` usage here https://github.com/src-d/ml/pull/145/files#diff-86f78a3692ae357bb5a7628fcf0cbc6bR171